### DISCUSSION TITLE
Failed to replace a string

### DISCUSSION THREAD

**Discussion by lvjr** [typst/typst#3875] [Q&A] [answered]
Hi all, I tried `str.replace` but failed to get correct result:
```typ
#let s = "1 2aaa b 3 dd ef 44 g d 6"
#let s = s.replace("\d", "")
#s
```
The `s` string didn't change. What is the problem with the above code? Thank you.

**Comment by frozolotl** [comment, marked as answer]
You want to specify that your needle is a regex; otherwise it will replace the literal string `\d`.
```typ
#let s = "1 2aaa b 3 dd ef 44 g d 6"
#let s = s.replace(regex("\d"), "")
#s
```

**Reply by lvjr** [reply]
Thank you very much!
